docs(css): Fx116 - syntax updates for offset-path and ray() (#28348)
mdn/content: files/en-us/web/css/box_value/index.md

---
title: <box>
slug: Web/CSS/box_value
page-type: css-type
---

{{CSSRef}}

> The **`<box>`** value type describes different aspects of an element's box model and represents how elements are rendered and positioned on a web page. The value type can applied to properties such as {{cssxref("transform-box")}} and {{cssxref("background-clip")}}. The value type is a keyword that represents a [box edge](/en-US/docs/Web/CSS/CSS_box_model/Introduction_to_the_CSS_box_model).

Author: Done! Thanks for the 1:1 discussion to explain and sort this out - `<box` vs `<box-edge>`. As discussed, dropping the `_value` from the slug. I've made the changes to the intro para with minor tweaks:

> The **`<box-edge>`** keywords represent a [box edge](/en-US/docs/Web/CSS/CSS_box_model/Introduction_to_the_CSS_box_model). The keywords are used to define different aspects of an element's box model and how elements are positioned and rendered on screen. The `<box-edge>` keyword values are the components of, but not limited to, the`<visual-box>`, `<layout-box>`, `<paint-box>`, `<coord-box>`, and `<geometry-box>` data types. These types are applied to properties such as {{cssxref("transform-box")}} and {{cssxref("background-clip")}}.

> The **`<box>`** value type describes different aspects of an element's box model and represents how elements are rendered and positioned on a web page. The value type can applied to properties such as {{cssxref("transform-box")}} and {{cssxref("background-clip")}}. The value type is a keyword that represents a [box edge](/en-US/docs/Web/CSS/CSS_box_model/Introduction_to_the_CSS_box_model).

Author: I realized that we're saying "`<box-edge>` keywords" and the "Values" actually lists the types so there was a mismatch.
Since we're calling the "Values" on the page as `<visual-box>`, `<layout-box>`, `<paint-box>`, `<coord-box>`, and `<geometry-box>` and the "Syntax" also describes these data types, I have updated "`<box-edge>` keywords" to "`<box-edge>` data types". The switched up description looks like this (also split into two paras):

> The **`<box-edge>`** value types represent a [box edge](/en-US/docs/Web/CSS/CSS_box_model/Introduction_to_the_CSS_box_model) keyword, such as [`content-box`](#content-box) and [`border-box`](#border-box). The box-edge keywords are used to define different aspects of an element's box model and how elements are positioned and rendered on screen.
> 
> The box-edge keywords are the components of, but not limited to, the data types `<visual-box>`, `<layout-box>`, `<paint-box>`, `<coord-box>`, and `<geometry-box>`. These types are applied to properties such as {{cssxref("transform-box")}} and {{cssxref("background-clip")}}.

## Syntax

```css
<visual-box> = content-box | padding-box | border-box /* also referred to as <box> */
<layout-box> = <box> | margin-box /* also referred to as <shape-box> */
<paint-box> = <box> | fill-box | stroke-box
<coord-box> = <box> | fill-box | stroke-box | view-box
<geometry-box> = <layout-box> | fill-box | stroke-box | view-box
```

### Values

A `<box>` can be of the type `<visual-box>`, `<layout-box>`, `<paint-box>`, or `<coord-box>`.

- `<visual-box>`

  - : Refers to the rectangular box generated for an element as seen by a user on a web page. It includes the element's padding and border. It excludes the margin area.

- `<layout-box>`

  - : Refers to the space occupied by an element, including its content, padding, border, and margin. This value type is used for layout and positioning purposes.

- `<paint-box>`

  - : Refers to the area within the layout box that is used to render the content visually. This includes the area where the element's background and borders are painted. It excludes the margin area.

> - `<coord-box>`
>   - : Refers to the coordinate box used for positioning and sizing an element within its parent container. It is used to control how content flows around the edges of the box. It excludes the margin area.
> 

Reviewer: ```suggestion
- `<coord-box>`
  - : Refers to the coordinate box used for positioning and sizing an element within its parent container. It is used to control how content flows around the edges of the box. It excludes the margin area. This value type is used for the {{cssxref("mask-origin")}} property
- `<geometry-box>`
  - : Defines the reference box for the basic shape or, if specified by itself, it causes the edges of the specified box, including any corner shaping (such as a {{cssxref("border-radius")}}), to be the clipping path. This value type is used for the {{cssxref("mask-clip")}} property and the SVG {{SVGAttr("clip-path")}} attribute.
```

Author: Thanks for adding `<geometry-box>`! I somehow completely missed adding an entry for it.

Btw, shouldn't [mask-origin](https://developer.mozilla.org/en-US/docs/Web/CSS/mask-origin) be listed with `<geometry-box>`?
I've changed the property listed with `<coord-box>` to `offset-path` (as per the new syntax, [also updated as part of this PR](https://pr28348.content.dev.mdn.mozit.cloud/en-US/docs/Web/CSS/offset-path)).

### Keywords

> The different `<box>` value types can be specified as one of the keyword values listed below:

Reviewer: ```suggestion
The different `<box-edge>` keywords are defined as follows:
```

- `margin-box`

  - : Refers to the edge of the margin of the box. If there is no margin on a side, then the value is the same as `border-box`. In SVG, `margin-box` is treated as `stroke-box`.

- `fill-box`

  - : Refers to the object bounding box in SVG. In CSS, `fill-box` is treated as `content-box`. It is used to wrap the content around the edges defined by the `coord-box` values.

- `stroke-box`

  - : Refers to the stroke bounding box in SVG. In CSS, `stroke-box` is treated as `border-box`. It is used to define the shape of the element when strokes are applied.

- `view-box`

  - : Refers to the nearest SVG viewport element's origin box. The origin box is a rectangle with the width and height of the initial SVG user coordinate system established by the {{svgattr("viewBox")}} attribute for that element. The origin box is positioned such that its top left corner is anchored at the coordinate system origin. In CSS, `view-box` treated as `border-box`.
    > **Note:** When the SVG viewport is not anchored at the origin, the origin box does not correspond to the SVG viewport.

## See also

> - Properties that use `<visual-box>` data type: {{cssxref("background-clip")}}

Reviewer: i researched the stuff in "values" before i got here.... oops. We should either include it in all  values or in none of the values, but not a mix.

Author: I removed them from "See also" and listed the properties with "Values" like you've suggested.
I could not find any property that uses `<paint-box>`.
